src/test/lib/concurrency/stress_test.cpp

```diff
@@ -695,4 +707,87 @@ TEST_F(StressTest, VisibilityOfInsertsBeingRolledBack) {
   }
 }
 
+/**
+ * Test that adding and modifying a the TableKeyConstraints of a table concurrently does not lead to 
+ * deadlocks or inconsistencies (e.g. duplicate constraints).
+ */
+TEST_F(StressTest, AddModifyTableKeyConstraintsConcurrently) {
+  // Create a table with multiple TableKeyConstraints
+  auto table = std::make_shared<Table>(
+      TableColumnDefinitions{{"a", DataType::Int, false}, {"b", DataType::Int, false}, {"c", DataType::Int, false}},
+      TableType::Data, std::nullopt, UseMvcc::Yes);
+  table->add_soft_constraint(TableKeyConstraint{{ColumnID{0}}, KeyConstraintType::PRIMARY_KEY});
+
+  table->append({1, 1, 1});
+  table->append({2, 2, 2});
+  table->append({3, 3, 1});
+
+  Hyrise::get().storage_manager.add_table("dummy_table", table);
+
+  /** Run multiple modifications and additions concurrently that could potentially lead to race conditions:
+   * - `UccDiscoveryPlugin::_validate_ucc_candidates`
+   * - `StaticTableNode::unique_column_combinations`
+   * - `Table::soft_key_constraints().clear()` to force a revalidation of the constraints. This does not happen in
+   * production, but we want to test the concurrent insertion of constraints.
+   * NOTE: We do not execute `Table::add_soft_constraint` because we do not have the knowledge whether or not the
+   * constraint already exists.
+   */
```

Comment:
could you adjust the comment with the recent changes? Also, we're never in production :)